liquidator/OneInchExchange.js

```diff
@@ -0,0 +1,135 @@
+const Web3 = require("web3");
+
+const erc20Abi = require("./abi/IERC20.json");
+
+const oneSplitAbi = require("./abi/OneSplit.json");
+const oneSplitAddress = "0xC586BeF4a0992C495Cf22e1aeEE4E446CECDee0E";
+
+// As defined in 1inch
+const ETH_ADDRESS = "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE";
+
+class OneInchExchange {
+  /**
+   * @notice Creates OneInchExchange client
+   * @param {Object} web3 Web3 instance
+   * */
+  constructor({ web3 }) {
+    this.web3 = web3;
+    this.web3.currentProvider.timeout = 120000;
+    this.oneSplitContract = new web3.eth.Contract(oneSplitAbi, oneSplitAddress);
+
+    this.toBN = web3.utils.toBN;
+  }
+
+  /**
+   * @notice Swaps token on one inch
+   * @param {string} fromToken Address of token to swap from
+   * @param {string} toToken Address of token to swap to.
+   * @param {string} amountWei String amount to swap, in Wei.
+   * @param {Object} options Web3 options to supply to send, e.g.
+   *      { from: '0x0...',
+            value: '1000',
+            gasPrice: '... }
+   */
+  async swap({ fromToken, toToken, amountWei }, options = {}) {
+    if (!options.from) {
+      throw new Error("Missing from key in options");
+    }
+
+    // Need to approve ERC20 tokens
+    if (fromToken !== ETH_ADDRESS) {
+      const erc20Contract = new this.web3.eth.Contract(erc20Abi, fromToken);
+      await erc20Contract.methods.approve(oneSplitAddress, amountWei).send({
```

Comment:
This call needs to get the current gas price from the `gasPriceClient` or it will cause major issues under network connection. Take a look at the `liquidator.js` implementation [here](https://github.com/UMAprotocol/protocol/blob/9120d52051473e630f2cb1e626c5385ca39e3283/liquidator/liquidator.js#L263) to see an example of how it's used.

Reply:
Why not use `web3.eth.getGasPrice()?`

Reply:
We could use the web3 version of this but we've had issues with it in the past so decided to roll our own. granted our own implementation also had some issues but at least we have a bit more control over it.